gpu4pyscf/dft/gen_grid.py

```diff
@@ -447,19 +399,19 @@ def arg_group_grids(mol, coords, box_size=GROUP_BOX_SIZE):
                 atom_coords.max(axis=0) + GROUP_BOUNDARY_PENALTY]
     # how many boxes inside the boundary
     boxes = ((boundary[1] - boundary[0]) * (1./box_size)).round().astype(int)
-    tot_boxes = numpy.prod(boxes + 2)
+    boxes = cupy.asarray(boxes)
```

Comment:
Consider to use numpy for these temporary arrays. Temporary cupy array can lead to more memory fragmentation